src/data/markdown/docs/02 javascript api/11 jslib/02 functional.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "functional"
+excerpt: "Functional testing with k6"
+---
+
+The `functional` module is an external JavaScript library that provides APIs to simplify functional testing in k6. 
+This library is inspired by ava, jest and jasmine. 
+
+<Blockquote mod='warning'>
+
+#### This library is rapidly evolving.
+
+This library is stable enough to be useful, but pay attention to the new versions released in jslib.k6.io. 
+
+This documentation is for the last version only. If you discover that some of the code below does not work, it most likely means that you are using an older version.
+
+</Blockquote>
+
+
+See [getting started with functional.js](/javascript-api/jslib/functional/getting-started-with-functional-js) for usage examples.
+
+More advanced examples can be found in the [examples section](/examples/functional-testing)
+
+
+| Function | Description |
+| -------- | ----------- |
+| [test(name, function)](/javascript-api/jslib/functional/test-name-function)  | Entry point for creating tests.  |
```

Comment:
Is there a particular reason why you opted for `test` over the BDD-style function names (`describe`, `when`, `it`) that most other js testing tools use?

Reply:
I was inspired by `ava` more than other libs. I agree with you and @MStoykov that `test` is too ambiguous. Maybe `describe` is better here :thinking: 
I'll create a repo for this library and we can discuss it there.

Reply:
> @simskij is our auto-deploy on PR not working anymore? 🤔

Our auto-deploy works, but it will only post the URL if the first build works, unfortunately. 😭 
It's either that or a new comment from the bot for every push, I'm afraid. But I'd be happy to change it to that if that's preferable.